### PR TITLE
Alllow to conditionally remove fields

### LINKED ISSUE #48: should be able to bypass remove hook
Inside `feathers-authentication` we [lookup the user](https://github.com/feathersjs/feathers-authentication/blob/master/src/services/local/index.js#L29-L44) in order to compare their password. If someone has an `remove('password')` as an after hook no one can log in. So we need a way to be able to conditionally bypass the `remove` hook.

### PR DESCRIPTION
**Pull request by daffl**
Like this:

```js
// synchronous hook
service.before({
  create: hooks.remove('test', hook => hook.params.remove)
});

/// asynchronous hook
service.before({
  create: hooks.remove('test', hook => new Promise(resolve => {
    setTimeout(() => resolve(hook.params.remove), 20);
  }))
});
```      

Closes #48

### CODE REVIEW

File: src/bundled.js [hunk 1]
```diff
@@ -7,18 +7,27 @@ export function remove(... fields) {
...
-    if(result) {
-      if(hook.method === 'find' || Array.isArray(result)) {
-        // data.data if the find method is paginated
-        (result.data || result).forEach(removeFields);
-      } else {
-        removeFields(result);
+    const result = hook.type === 'before' ? hook.data : hook.result;
+    const next = condition => {
+      if(result && condition) {
+        if(hook.method === 'find' || Array.isArray(result)) {
```
Comment (ekryski) [inline src/bundled.js]: Should probably be able to remove the `hook.method === 'find'` part. we really only care whether it is an Array or not.

**Comment by ekryski** [comment]
Oh man this is sweet. Just going to pull it down and test with a running app with auth.

File: src/bundled.js (hunk 1 above)
Comment (daffl) [inline src/bundled.js]: I think it needs to stay because the `find` method can return a paginated object which is not an array.

Comment (ekryski) [inline src/bundled.js]: Quite right my friend.

**Comment by ekryski** [comment]
Booya! :shipit:

**Pull request merged**